coderd/audit.go

```diff
@@ -181,9 +184,9 @@ func (api *API) convertAuditLogs(ctx context.Context, dblogs []database.GetAudit
 }
 
 type AdditionalFields struct {
-	WorkspaceName string               `json:"workspace_name"`
-	BuildNumber   string               `json:"build_number"`
-	BuildReason   database.BuildReason `json:"build_reason"`
+	WorkspaceName string
+	BuildNumber   string
+	BuildReason   database.BuildReason
```

Comment:
@Emyrk in a previous PR, you advised me to tag these. Doing so caused them to not unmarshal correctly [here](https://github.com/coder/coder/pull/5895/files#diff-27c456ef04dc75c55ae773fdff8c7a05c9ecaf54691704f0ca0425a6c5300dc7L214). Do you have any idea what I screwed up? I wasn't quite sure so I've just removed the tags for now and added a test.

Reply:
I was unaware we write additional fields like this: https://github.com/coder/coder/blob/7d0fad4f389f77e83076465cbb763c028ef918dc/coderd/provisionerdserver/provisionerdserver.go#L566-L575

We should really be using the same type on both sides of the marshal, rather than using this map on 1, and a struct on another.

Any change though would break backwards compatibility, unless we added something to catch the old field names too.

Reply:
I suggest we change that code to this:
```golang
// TODO: Where should we import AdditionalFields from? Where should this be defined?
buildResourceInfo := external.AdditionalFields{
	WorkspaceName: workspace.Name,
	BuildNumber:   strconv.FormatInt(int64(build.BuildNumber), 10),
	BuildReason:   database.BuildReason(fmt.Sprintf("%v", build.Reason)),
}

wriBytes, err := json.Marshal(buildResourceInfo)
if err != nil {
	server.Logger.Error(ctx, "marshal workspace resource info for failed job", slog.Error(err))
}
```

Maybe `AdditionalFields` should be a struct defined in `database/`.

Reply:
Stuck it in `coderd/audit/audit.go`!

Reply:
Nice!! Is backwards compatibly required? Because there is a way to do that if it is

Reply:
I don't think it's required, but also - I think it's already working somehow:
![Screen Shot 2023-01-27 at 9 16 18 PM](https://user-images.githubusercontent.com/19142439/215236961-0da22c19-4d58-4751-b3d6-c55e5eecb0b6.png)

So we should be good to go!